homeassistant/components/iotty/switch.py

```diff
@@ -0,0 +1,110 @@
+"""Implement a iotty Light Switch Device."""
+import logging
+from typing import Any
+
+from iottycloud.lightswitch import LightSwitch
+from iottycloud.verbs import LS_DEVICE_TYPE_UID
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .api import IottyProxy
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class IottyLightSwitch(SwitchEntity):
+    """Haas entity class for iotty LightSwitch."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_device_class = SwitchDeviceClass.SWITCH
+    _iotty_cloud: IottyProxy
+    _iotty_device: LightSwitch
+
+    def __init__(self, iotty_cloud: IottyProxy, iotty_device: LightSwitch) -> None:
+        """Initialize the LightSwitch device."""
+        super().__init__()
+
+        if iotty_cloud is None:
+            raise ValueError("iotty_cloud")
+
+        if iotty_device is None:
```

Comment:
Can this ever happen?

Reply:
Lines 35-36, were already removed to address another comment. Do you mean line 38?